app/controllers/declaration/RepresentativeDetailsPageController.scala

```diff
@@ -62,8 +63,15 @@ class RepresentativeDetailsPageController @Inject()(
         validRepresentativeDetails =>
           customsCacheService
             .cache[RepresentativeDetails](cacheId, RepresentativeDetails.formId, validRepresentativeDetails)
-            .map(_ => Redirect(controllers.declaration.routes.DeclarationAdditionalActorsController.displayForm()))
+            .map(
+              _ =>
+                request.choice.value match {
```

Comment:
May I suggest separating this pattern matching into a private method?
We have a similar case done in `DeclarationTypeController`

Reply:
Awesome! Done.